clients/src/main/java/org/apache/kafka/clients/admin/ScramMechanism.java

```diff
@@ -49,8 +53,11 @@ public static ScramMechanism fromType(byte type) {
      *     Salted Challenge Response Authentication Mechanism (SCRAM) SASL and GSS-API Mechanisms, Section 4</a>
      */
     public static ScramMechanism fromMechanismName(String mechanismName) {
-        ScramMechanism retvalFoundMechanism = ScramMechanism.valueOf(mechanismName.replace('-', '_'));
-        return retvalFoundMechanism != null ? retvalFoundMechanism : UNKNOWN;
+        String normalizedMechanism = mechanismName.replace('-', '_');
+        return Arrays.stream(VALUES)
+            .filter(mechanism -> mechanism.name().equals(normalizedMechanism))
+            .findFirst()
+            .orElse(UNKNOWN);
```

Comment:
@rondagostino looks like the original implementation didn't achieve the goal. I added a test that failed without this change.

Reply:
Ah, you are right: `valueOf()` `Throws: IllegalArgumentException - if the specified enum type has no constant with the specified name, or the specified class object does not represent an enum type.`

Thanks for catching/testing for/fixing it.

Reply:
@rondagostino do we need to cherry-pick this change to the 2.7 branch? I am not sure under which context this method is used and whether it's important enough to backport.